openvpn.c

```diff
@@ -2177,18 +2160,53 @@ StartOpenVPN(connection_t *c)
         }
         return FALSE;
     }
-    PrintDebug(L"Starting openvpn on config %ls", c->config_name);
+    else if (c->state != disconnected)
+    {
+        return FALSE;
+    }
 
-    RunPreconnectScript(c);
+    PrintDebug(L"Starting openvpn on config %ls", c->config_name);
 
     /* Create thread to show the connection's status dialog */
-    hThread = CreateThread(NULL, 0, ThreadOpenVPNStatus, c, CREATE_SUSPENDED, &c->threadId);
+    HANDLE hThread = CreateThread(NULL, 0, ThreadOpenVPNStatus, c, CREATE_SUSPENDED, &c->threadId);
     if (hThread == NULL)
     {
         ShowLocalizedMsg(IDS_ERR_CREATE_THREAD_STATUS);
-        goto out;
+        return false;
+    }
+
+    /* Launch openvpn.exe using the service or directly */
+    if (!LaunchOpenVPN(c))
+    {
+        TerminateThread(hThread, 1);
```

Comment:
Is this new `TerminateThread()` call really needed?

Reply:
If we fail here, won't the thread created above hang around in a suspended state and leak memory?

Reply:
Yes, its a bug-fix as well.